python/pyarrow/src/arrow/python/pythoncapi_compat.h

```diff
@@ -0,0 +1,1513 @@
+// Header file providing new C API functions to old Python versions.
+//
+// File distributed under the Zero Clause BSD (0BSD) license.
+// Copyright Contributors to the pythoncapi_compat project.
+//
+// Homepage:
+// https://github.com/python/pythoncapi_compat
+//
+// Latest version:
+// https://raw.githubusercontent.com/python/pythoncapi_compat/master/pythoncapi_compat.h
+//
+// SPDX-License-Identifier: 0BSD
+
```

Comment:
Can you add a comment somewhere stating which upstream changeset this is vendored from?

Reply:
Done.